src/main/java/edu/harvard/iq/dataverse/engine/command/impl/GetLatestAccessibleFileMetadataCommand.java

```diff
@@ -0,0 +1,41 @@
+package edu.harvard.iq.dataverse.engine.command.impl;
+
+import edu.harvard.iq.dataverse.DataFile;
+import edu.harvard.iq.dataverse.FileMetadata;
+import edu.harvard.iq.dataverse.authorization.Permission;
+import edu.harvard.iq.dataverse.engine.command.AbstractCommand;
+import edu.harvard.iq.dataverse.engine.command.CommandContext;
+import edu.harvard.iq.dataverse.engine.command.DataverseRequest;
+import edu.harvard.iq.dataverse.engine.command.RequiredPermissions;
+import edu.harvard.iq.dataverse.engine.command.exception.CommandException;
+
+@RequiredPermissions({})
+public class GetLatestAccessibleFileMetadataCommand extends AbstractCommand<FileMetadata> {
+    private final DataFile dataFile;
+    private final boolean includeDeaccessioned;
+
+    public GetLatestAccessibleFileMetadataCommand(DataverseRequest request, DataFile dataFile, boolean includeDeaccessioned) {
+        super(request, dataFile);
+        this.dataFile = dataFile;
+        this.includeDeaccessioned = includeDeaccessioned;
+    }
+
+    @Override
+    public FileMetadata execute(CommandContext ctxt) throws CommandException {
+        FileMetadata fileMetadata = null;
+
+        if (ctxt.permissions().requestOn(getRequest(), dataFile.getOwner()).has(Permission.ViewUnpublishedDataset)) {
+            fileMetadata = ctxt.engine().submit(
+                    new GetDraftFileMetadataIfAvailableCommand(getRequest(), dataFile)
```

Comment:
this is going to go through all the command setup and recheck the permission, just to call DataFile.getLatestFileMetadata() and see if it's draft - would it be easier/more efficient to just repeat those two lines here (or even make a 1 line DataFile.getDraftFileMetadata() call to use in both?

Reply:
Makes sense, I just added `DataFile.getDraftFileMetadata()`. Thanks.